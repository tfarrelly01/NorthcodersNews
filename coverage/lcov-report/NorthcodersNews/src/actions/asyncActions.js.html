<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for NorthcodersNews/src/actions/asyncActions.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">NorthcodersNews/src/actions</a> asyncActions.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import axios from 'axios';
import { ROOT } from '../../config';
import { 
  fetchTopicsRequest, 
  fetchTopicsSuccess, 
  fetchTopicsError,
} from './actionCreators';
&nbsp;
// GET api/topics
export const fetchTopics = () =&gt; {
  return dispatch =&gt; {
    dispatch(fetchTopicsRequest());
    axios.get(`${ROOT}/topics`)
      .then(res =&gt; {
        dispatch(fetchTopicsSuccess(res.data.topics));
      })
      .catch(<span class="fstat-no" title="function not covered" >er</span>r =&gt; {
<span class="cstat-no" title="statement not covered" >        dispatch(fetchTopicsError(err));</span>
      });
&nbsp;
  };
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Sep 29 2017 13:00:02 GMT+0100 (BST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
