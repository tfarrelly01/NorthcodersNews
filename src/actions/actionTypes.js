export const FETCH_TOPICS_REQUEST = 'FETCH_TOPICS_REQUEST';
export const FETCH_TOPICS_SUCCESS = 'FETCH_TOPICS_SUCCESS';
export const FETCH_TOPICS_ERROR = 'FETCH_TOPICS_ERROR';

export const FETCH_ARTICLES_REQUEST = 'FETCH_ARTICLES_REQUEST';
export const FETCH_ARTICLES_SUCCESS = 'FETCH_ARTICLES_SUCCESS';
export const FETCH_ARTICLES_ERROR = 'FETCH_ARTICLES_ERROR';

export const FETCH_USERS_REQUEST = 'FETCH_USERS_REQUEST';
export const FETCH_USERS_SUCCESS = 'FETCH_USERS_SUCCESS';
export const FETCH_USERS_ERROR = 'FETCH_USERS_ERROR';

