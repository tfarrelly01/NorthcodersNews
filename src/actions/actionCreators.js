import * as types from './actionTypes';

export function fetchTopicsRequest () {
  return {
    type: types.FETCH_TOPICS_REQUEST
  };
}

export function fetchTopicsSuccess (topics) {
  return {
    type: types.FETCH_TOPICS_SUCCESS,
    payload: topics
  };
}

export function fetchTopicsError (error) {
  return {
    type: types.FETCH_TOPICS_ERROR,
    payload: error
  };
}

export function fetchArticlesRequest () {
  return {
    type: types.FETCH_ARTICLES_REQUEST
  };
}

export function fetchArticlesSuccess (articles) {
  return {
    type: types.FETCH_ARTICLES_SUCCESS,
    payload: articles
  };
}

export function fetchArticlesError (error) {
  return {
    type: types.FETCH_ARTICLES_ERROR,
    payload: error
  };
}