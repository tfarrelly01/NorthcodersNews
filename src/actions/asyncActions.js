import axios from 'axios';
import { ROOT } from '../../config';
import * as actions from './actionCreators';

// GET api/topics
export const fetchTopics = () => {
  return dispatch => {
    dispatch(actions.fetchTopicsRequest());
    axios.get(`${ROOT}/topics`)
      .then(res => {
        dispatch(actions.fetchTopicsSuccess(res.data.topics));
      })
      .catch(err => {
        dispatch(actions.fetchTopicsError(err));
      });
  };
};

// GET api/articles
export const fetchArticles = () => {
  return dispatch => {
    dispatch(actions.fetchArticlesRequest());
    axios.get(`${ROOT}/articles`)
      .then(res => {
        dispatch(actions.fetchArticlesSuccess(res.data.articles));
      })
      .catch(err => {
        dispatch(actions.fetchArticlesError(err));
      });
  };
};

// GET api/users
export const fetchUsers = () => {
  return dispatch => {
    dispatch(actions.fetchUsersRequest());
    axios.get(`${ROOT}/users`)
      .then(res => {
        dispatch(actions.fetchUsersSuccess(res.data.users));
      })
      .catch(err => {
        dispatch(actions.fetchUsersError(err));
      });
  };
};
