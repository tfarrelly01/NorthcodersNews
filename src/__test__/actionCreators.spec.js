import * as actions from '../actions/actionCreators';
import  { fetchTopics } from '../actions/asyncActions';

const data = [
  'foo',
  'bar',
  'baz'
];
const errMsg = 'ERROR - Error message';

describe('Test Action Creators ', () => {

	describe('fetchTopics', () => {
		test('is a function', () => {
			expect(typeof fetchTopics).toBe('function');
		});

		test('request action returns an object', () => {
			expect(typeof actions.fetchTopicsRequest()).toEqual('object');
		});

		test('success action returns an object', () => {
			expect(typeof actions.fetchTopicsSuccess()).toEqual('object');
		});

		test('error action returns an object', () => {
			expect(typeof actions.fetchTopicsError()).toEqual('object');
		});

		test('success action returns data passed', () => {
			const test = actions.fetchTopicsSuccess(data);
      expect(test).toMatchSnapshot();
		});

		test('error action returns data passed', () => {
			const test = actions.fetchTopicsError(errMsg);
      expect(test.payload).toEqual(errMsg);
		});
	});

});
