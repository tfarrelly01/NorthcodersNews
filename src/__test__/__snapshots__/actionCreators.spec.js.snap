// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Action Creators  Action Creator: fetchArticle success action returns data passed 1`] = `
Object {
  "payload": Array [
    "foo",
    "bar",
    "baz",
  ],
  "type": "FETCH_ARTICLE_SUCCESS",
}
`;

exports[`Test Action Creators  Action Creator: fetchArticles success action returns data passed 1`] = `
Object {
  "payload": Array [
    "foo",
    "bar",
    "baz",
  ],
  "type": "FETCH_ARTICLES_SUCCESS",
}
`;

exports[`Test Action Creators  Action Creator: fetchTopicArticles success action returns data passed 1`] = `
Object {
  "payload": Array [
    "foo",
    "bar",
    "baz",
  ],
  "type": "FETCH_TOPIC_ARTICLES_SUCCESS",
}
`;

exports[`Test Action Creators  Action Creator: fetchTopics success action returns data passed 1`] = `
Object {
  "payload": Array [
    "foo",
    "bar",
    "baz",
  ],
  "type": "FETCH_TOPICS_SUCCESS",
}
`;

exports[`Test Action Creators  Action Creator: fetchUsers success action returns data passed 1`] = `
Object {
  "payload": Array [
    "foo",
    "bar",
    "baz",
  ],
  "type": "FETCH_USERS_SUCCESS",
}
`;
